octopoes/octopoes/models/types.py

```diff
@@ -205,8 +205,15 @@ def to_concrete(object_types: set[type[OOI]]) -> set[type[OOI]]:
     return concrete_types
 
 
+class TypeNotFound(Exception):
+    pass
+
+
 def type_by_name(type_name: str):
-    return next(t for t in ALL_TYPES if t.__name__ == type_name)
+    try:
+        return next(t for t in ALL_TYPES if t.__name__ == type_name)
+    except StopIteration:
+        raise TypeNotFound
```

Comment:
Cool 👍  Shall I extend the `ValueError` in the `TypeNotFound`? I think perhaps having some more custom exceptions allows more granularity in the rest of OpenKat?

Reply:
Absolutely! Extending `ValueError` in this case is appropriate. It will offer more granularity in the codebase and makes error handling more specific and meaningful 👍